### PR TITLE
Initial gadget template

### PR DESCRIPTION
**Pull request by alban**
# Initial gadget template

This is part of https://github.com/inspektor-gadget/inspektor-gadget/issues/2382

## How to use

README.md explains how to use it

## Testing done

```
make build
make clang-format
```

## Out of scope

This initial PR does not include:
- Tests and CI
- Publishing a gadget on ghcr.io from a GitHub Action
- A reusable GitHub Action

This could be added in future PRs.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+CLANG_FORMAT ?= clang-format
+
+.PHONY: build
+build:
+	IG_EXPERIMENTAL=true sudo -E \
+		ig image build \
+			-t $(CONTAINER_REPO):$(IMAGE_TAG) \
+			--update-metadata .
+
+.PHONY: run
```
Comment (blanquicet) [inline Makefile:13]:
```suggestion
# PARAMS can be used to pass additional parameters locally. For example:
# PARAMS="-o jsonpretty" make run
.PHONY: run
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+IMAGE_TAG ?= latest
+
+CLANG_FORMAT ?= clang-format
+
+.PHONY: build
+build:
+	IG_EXPERIMENTAL=true sudo -E \
+		ig image build \
+			-t $(CONTAINER_REPO):$(IMAGE_TAG) \
+			--update-metadata .
```
Comment (blanquicet) [inline Makefile:11]: Why always using `--update-metadata`? 

Not related to this template but IG framework: I'm wondering if we should enable this feature by default.

File: gadget.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    fields:
+    - name: timestamp
+      attributes:
+        template: timestamp
+    - name: pid
+      attributes:
+        template: pid
+    - name: comm
+      attributes:
+        template: comm
```
[the remark this answers is not in the document]
Comment (alban) [reply gadget.yaml:18]: I thought that it was not necessary when there is a template.

[the remark this answers is not in the document]
Comment (blanquicet) [reply gadget.yaml:18]: I prefer having examples that are following the best-practices :)

[the remark this answers is not in the document]
Comment (mauriciovasquezbernal) [reply gadget.yaml:18]: I think you both are talking about different kind of "template" :D. 

@alban the field template doesn't include the description.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,25 @@
+CONTAINER_REPO ?= ghcr.io/changeme-org/changeme-gadget-name
+IMAGE_TAG ?= latest
```
Comment (mauriciovasquezbernal) [inline Makefile:2]: Perhaps good place to use `git describe --tags --always --dirty`?

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+CLANG_FORMAT ?= clang-format
+
+.PHONY: build
+build:
+	IG_EXPERIMENTAL=true sudo -E \
```
Comment (mauriciovasquezbernal) [inline Makefile:8]: Is it possible to move `IG_EXPERIMENTAL=true` at the top of the makefile?

**Review by mauriciovasquezbernal — APPROVED** [review]
It looks great. Feel free to merge when you want.

File: gadget.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+name: 'TODO: Fill the gadget name'
+description: 'TODO: Fill the gadget description'
+tracers:
+  events:
```
[the remark this answers is not in the document]
Comment (alban) [reply gadget.yaml:4]: When I try your change, `--update-metadata` reverts back to `events` .

File: Makefile (hunk 2 above)
Comment (blanquicet) [reply Makefile]: I think this is an issue, it should update it with `changeme_mygadget` from the eBPF code. Let me check this upstream.

File: gadget.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (blanquicet) [reply gadget.yaml:4]: https://github.com/inspektor-gadget/gadget-template/pull/2

**Pull request merged**